### PR TITLE
Add --additional_runtime_args option and use in OPT example.

### PR DESCRIPTION
**Pull request by monorimet**
Fix the func name. (#1838)

fix non-plugin path

fixup OPT example

Use compile_file

### CODE REVIEW

File: tank/examples/opt/opt_perf_comparison.py [hunk 1]
```diff
@@ -96,39 +119,53 @@ def create_vmfb_module(
...
-            f.write(model_mlir)
-        print(f"Saved mlir at {mlir_path}")
-
-    shark_module = SharkInference(
-        model_mlir,
-        device=device,
-        mlir_dialect="tm_tensor",
-        is_benchmark=False,
-    )
+    # with open(mlir_path, "r") as f:
```
Comment (dan-garvey) [inline tank/examples/opt/opt_perf_comparison.py:129]: dead code?

File: tank/examples/opt/opt_perf_comparison.py [hunk 2]
```diff
@@ -96,39 +119,53 @@ def create_vmfb_module(
...
+    #    rt_flags=[],
+    # )
 
     vmfb_name = (
         f"{opt_fs_name}_causallm_{max_seq_len}_torch_{DEVICE}_tiled_ukernels"
     )
-    shark_module.save_module(module_name=vmfb_name)
+    # shark_module.save_module(module_name=vmfb_name)
     vmfb_path = vmfb_name + ".vmfb"
+    ireec.compile_file(
```
Comment (dan-garvey) [inline tank/examples/opt/opt_perf_comparison.py:146]: we dont have an abstraction for this?

**Review by dan-garvey — CHANGES_REQUESTED** [review]

File: tank/examples/opt/opt_perf_comparison.py (hunk 2 above)
Comment (monorimet) [reply tank/examples/opt/opt_perf_comparison.py]: This was proof of concept for the new path through shark inference (compile_str --> compile_file).
Updated it, thanks!

File: tank/examples/opt/opt_perf_comparison.py (hunk 1 above)
Comment (monorimet) [reply tank/examples/opt/opt_perf_comparison.py]: See next comment :)

**Review by dan-garvey — APPROVED** [review]

**Pull request merged**